### ISSUE TITLE
Output of Profile.retrieve() doesn't unserialize in 0.5.1

### ISSUE THREAD

**Issue by keatonmill** [JuliaLang/julia#21025]
The following code results in an InexactError on the newest Linux binary:

```julia
M = Matrix{Float64}[rand(1000,1000) for i=1:10]

Profile.clear()
@profile map(svd, M)

r = Profile.retrieve()

Profile.print(r[1],r[2])

f = open("profile.bin", "w")
serialize(f,r)
close(f)

s = deserialize(open("profile.bin")) # error here
Profile.print(s[1], s[2])
```

`versioninfo()` reports:
```
Julia Version 0.5.1
Commit 6445c82 (2017-03-05 13:25 UTC)
Platform Info:
  OS: Linux (x86_64-pc-linux-gnu)
  CPU: Intel(R) Xeon(R) CPU E5-2640 v4 @ 2.40GHz
  WORD_SIZE: 64
  BLAS: libopenblas (USE64BITINT DYNAMIC_ARCH NO_AFFINITY Haswell)
  LAPACK: libopenblas64_
  LIBM: libopenlibm
  LLVM: libLLVM-3.7.1 (ORCJIT, broadwell)
```

The error I receive is:
```
ERROR: LoadError: InexactError()
 in deserialize(::SerializationState{IOStream}, ::Type{StackFrame}) at ./stacktraces.jl:113
 in deserialize_datatype(::SerializationState{IOStream}) at ./serialize.jl:837
 in handle_deserialize(::SerializationState{IOStream}, ::Int32) at ./serialize.jl:571
 in deserialize_array(::SerializationState{IOStream}) at ./serialize.jl:711
 in handle_deserialize(::SerializationState{IOStream}, ::Int32) at ./serialize.jl:569
 in deserialize(::SerializationState{IOStream}, ::Type{Dict{UInt64,Array{StackFrame,1}}}) at ./serialize.jl:896
 in deserialize_datatype(::SerializationState{IOStream}) at ./serialize.jl:837
 in handle_deserialize(::SerializationState{IOStream}, ::Int32) at ./serialize.jl:571
 in deserialize(::SerializationState{IOStream}) at ./serialize.jl:541
 in ntuple(::Base.Serializer.##1#2{SerializationState{IOStream}}, ::Int64) at ./tuple.jl:65
 in handle_deserialize(::SerializationState{IOStream}, ::Int32) at ./serialize.jl:562
 in deserialize(::IOStream) at ./serialize.jl:538
 in include_from_node1(::String) at ./loading.jl:488
while loading /home/user/simple-profile-test.jl, in expression starting on line 14
```

**Comment by tkelman** [comment]
So this was broken by https://github.com/JuliaLang/julia/commit/e5f7f5cf344968bc17378e27d212a9d405fe7ced which changed a field type to unsigned in StackFrame. cc @vtjnash

**Comment by tkelman** [comment]
Looks like it should be a fairly easy fix, change https://github.com/JuliaLang/julia/blob/v0.5.1/base/stacktraces.jl#L112 and add a test. Was fixed on 0.6 by https://github.com/JuliaLang/julia/pull/18043 which had a bunch of other API changes so the rest of that isn't a backport candidate.

**Comment by tkelman** [comment]
fixed on master, test added in #21575, will be backported for 0.5.2 (https://github.com/JuliaLang/julia/pull/21684)
